cartographer_rviz/cartographer_rviz/submaps_display.cc

```diff
@@ -149,6 +162,53 @@ void SubmapsDisplay::update(const float wall_dt, const float ros_dt) {
   }
 }
 
+void SubmapsDisplay::AllEnabledChanged() {
+  ::cartographer::common::MutexLocker locker(&mutex_);
+  bool visibility = visibility_all_enabled_->getBool();
+  for (auto &trajectory : trajectories_) {
+    for (auto &submap : trajectory) {
+      DisconnectVisibilityChange(submap);
+      submap->SetVisibility(visibility);
+      ConnectVisibilityChange(submap);
+    }
+  }
+}
+
+void SubmapsDisplay::VisibilityChanged(DrawableSubmap* submap) {
```

Comment:
Edit: In the end, I added it only to line 177 as you originally said. Qt won't connect a signal with a pointer to non-const to a slot which receives a pointer to const.